### ISSUE TITLE
Get role from the user table

### ISSUE THREAD

**Issue by lishine** [hasura/graphql-engine#2638]
Related:
#2328 
#2512 
#2297 

TLDR implement permission logic for all tables - for reading the user role and other stuff.
What I understand is that role is brought to Hasura externally but my understanding is that role is stored in db, so why not read it automatically upon query (The usual approach is to cache it in JWT).

In my case
1. Hasura is accessed only by backend api on the same server. Client's queries are forwarded through the api.
2. API recieves only user id from the client, then queries the role from Hasura User table and then attaches this role to the request with query. This is done for each access to the API in order to get the most up to date role for the user, in contrast to the usual way of storing the role in JWT.

I am thinking whether it is possible for Hasura to take the role from User table automatically, so as time won't wasted for another request to get the role.
This can be generalized to some logic that it is invoked practically on each access to any table - read the role for example and read the user status - maybe it is denied of access and if it exists at all!

**Comment by revskill10** [comment]
I think it depends.
In my app, for example, there's no Roles table.

But there're people table like Teacher, Students,..., so user's role is determined based on user's properties dynamically.
